### ISSUE TITLE
Please split minkindr_python into separate repo!

### ISSUE THREAD

**Issue by helenol** [ethz-asl/minkindr#69]
Currently mindkindr_python is in this repo, which is meant to be as minimal implementation of a transformation library as possible. It brings in huge dependencies like numpy_eigen, which takes ~20-30 minutes to build on a normal machine and forces everyone using minkindr (again, designed as an absolutely _minimal_ transformation package) to install numpy_eigen and catkin_boost_python_buildtool. It's breaking default installation instructions of packages such as voxblox, as it now adds many additional dependencies that don't allow an entire workspace to build.
This is also an issue for setting up continuous integration pipelines, as again, many dependencies are introduced and the entire workspace can no longer be built without them.

Great work on introducing python bindings, but _please put them outside this repo_.

@HannesSommer @mbuerki @alexmillane @michaelpantic @mfehr

**Comment by michaelpantic** [comment]
Fair point! I don't have anything to do with minkindr itself, but you tagged me soooo there's my opinion :-D

I think it's an equally fair point to keep minkindr and its python bindings in the same repo. Also they are tightly coupled in the same release cycle, so it makes sense to keep it in the same repo. 

How about the following: We keep it in the same repo, but minkindr_python checks if dependencies are installed and if not, just skips the build? Shouldn't be too much effort in the cmakelists file.

**Comment by michaelpantic** [comment]
ah, that was already talked about here https://github.com/ethz-asl/minkindr/pull/56#issuecomment-435482595 but forgotten ;-)

**Comment by mfehr** [comment]
Here are my 2 cents: Agree with @michaelpantic, for the same reasons. Regarding the solutions suggested by @HannesSommer  in [#56 (comment)](https://github.com/ethz-asl/minkindr/pull/56#issuecomment-435482595),  I would therefore go for option 3, the CMAKE magic.

**Comment by HannesSommer** [comment]
@helenol, I don't think your points are that convincing:
* I have still quite vivid memories of when Paul created minikindr. The "minimal" was not about dependencies of the repo but instead about a minimal interface and feature set of the package itself in particular in comparison to kindr. in fact he typically was putting python exports into the same repo and would probably have done as it is now if python had been a requirement back then
* numpy_eigen's extreme built times are long history: building numpy_eigen takes < 40s on a 6 core i5 with no HT (i5-8500) and [my last compile time opt](https://github.com/ethz-asl/schweizer_messer/commit/172953b1053fae7f717b7ec3cc93f9c9b42b8a9f) for numpy is more than 2 years old. How can you still experience such built times?
* entire work spaces can very well be built since there are many ways to prevent an unwanted package to build as part of that: `CATKIN_IGNORE`, `catkin config --blacklist`, and simply deleting are options which work very well in particular in a CI context. Besides, I would recommend not to build entire work spaces per default because. It wastes time and energy all the time. Have you considered using `--this` or `catkin catkin config --whitelist X Y Z`? 
* CI is typically configured to compile all packages of the primary repo but not all packages in the dependency repositories. Otherwise one would waste precious CI time and who has time to wait for CI? For the CI of minkindr it makes total sense to build the python package as well. If the problem is on ASL's Jenkins I'm happy to fix it there for you
* Voxblox's build instructions [here](https://voxblox.readthedocs.io/en/latest/pages/Installation.html#installation) do suggest to build it with `cd ~/catkin_ws/src/
catkin build voxblox_ros`, i.e. not the entire workspace. How can this be affected by the separate `minkindr_python` package? Did I overlook something? In any case, I doubt that changing a build instruction once is that much of a cost.

On the pro side I see the following: that packages that are tightly coupled like the two in this repo benefit from being in one repo because sharing CI and PRs makes synchronous changes much nicer and easier for the developers of the libraries, it is harder to overlook when one breaks the python export by chaining the core library, and setting up and maintaining another CI is also some work. Furthermore it is a breaking change for the few that already depend on the python export being in this library.

I don't see how your arguments outweigh these arguments for keeping in one repo. I would therefore prefer to either keep as is or try the CMake trick @michaelpantic mentions  -- if somebody has the time to do it. I would be glad to review a PR for it (first step would be to remove the `ALL_DEPS_REQUIRED` [here](https://github.com/ethz-asl/minkindr/blob/master/minkindr_python/CMakeLists.txt#L5))

But I'm also not super strongly against a split -- again, if somebody has the time to do all that is necessary to get a somewhat equivalent setup.
